api/v1/apicurioregistry_types.go

```diff
@@ -100,18 +100,149 @@ type ApicurioRegistrySpecDeploymentMetadata struct {
 }
 
 type ApicurioRegistrySpecDeployment struct {
-	Replicas    int32               `json:"replicas,omitempty"`
-	Host        string              `json:"host,omitempty"`
-	Affinity    *corev1.Affinity    `json:"affinity,omitempty"`
-	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+	Replicas    int32             `json:"replicas,omitempty"`
+	Host        string            `json:"host,omitempty"`
+	Affinity    *core.Affinity    `json:"affinity,omitempty"`
+	Tolerations []core.Toleration `json:"tolerations,omitempty"`
 	// Metadata applied to the Deployment pod template.
 	Metadata ApicurioRegistrySpecDeploymentMetadata `json:"metadata,omitempty"`
 	// Image set in the Deployment pod template. Overrides the values in the REGISTRY_IMAGE_MEM, REGISTRY_IMAGE_KAFKASQL and REGISTRY_IMAGE_SQL operator environment variables.
 	Image string `json:"image,omitempty"`
 	// List of secrets in the same namespace to use for pulling the Deployment pod image.
-	ImagePullSecrets []corev1.LocalObjectReference `json:"imagePullSecrets,omitempty"`
+	ImagePullSecrets []core.LocalObjectReference `json:"imagePullSecrets,omitempty"`
 	// Configure how the Operator manages Kubernetes resources
 	ManagedResources ApicurioRegistrySpecDeploymentManagedResources `json:"managedResources,omitempty"`
+
+	PodSpecPreview ApicurioRegistrySpecDeploymentPodSpecPreview `json:"podSpecPreview,omitempty"`
```

Comment:
If I'm not wrong we agreed on moving on with [`PodTemplate`](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#podtemplatespec-v1-core) instead of [`PodSpec`](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#podspec-v1-core) as the former includes the possibility to customize `labels`/`annotation` which is critical to enable the usage of custom network policies, service meshes etc.

Reply:
Done